lib/persistence/http/RestClient.dart

```diff
@@ -0,0 +1,21 @@
+import 'package:coffeecard/model/AppConfig.dart';
+import 'package:coffeecard/model/Login.dart';
+import 'package:coffeecard/model/Token.dart';
+import 'package:dio/dio.dart';
+import 'package:retrofit/retrofit.dart';
+
+part 'RestClient.g.dart';
+
+@RestApi()
+abstract class RestClient {
+  static const String PRODUCTION_URL = "https://analogio.dk/clippy";
+  static const String DEBUG_URL = "https://beta.analogio.dk/api/clippy";
+
+  factory RestClient(Dio dio, {String baseUrl}) = _RestClient;
+
+  @GET("​/api​/v1​/AppConfig")
```

Comment:
I personally don't like that we will end up having every single endpoint in the rest-client. Is this really the recommended way to do it?

Reply:
That's unfortunate :/ How much boilerplate code are we avoiding by using retrofit instead of using dio directly, where we have a bit more freedom?

Reply:
Retrofit generates a lot of boilerplate for us (it can easily be some hundred lines) in the long term. I think it's quite neat. I'm not sure what we are missing and as we can make custom Dio factories we are able to override certain behaviour if needed.

Reply:
Try to run the script locally and take a look in what is generated just to give you an idea :-)

Reply:
Only thing I was missing was not having the methods in the rest client, but I suppose I can live with it :/

Reply:
Haha :sweat_smile: Well, thats the Retrofit core, so that's how it is

Reply:
I have renamed the RestClient to `CoffeeCardApiClient` to emphasize that it is not a generic client but a specific one for the CoffeeCard API.